game/client/gui_helpers/stack_menu.lua

```diff
@@ -23,12 +25,52 @@ function StackMenu:create(yStart, overlay, gap, title)
     return vals
 end
 
-function StackMenu:nextWidgetPosition()
-    local position = getWidgetCentre(WIDGET_WIDTH, self.y)
-    self.y = self.y - self.widgetGap
+function StackMenu:nextWidgetPosition(width, centered, x, y)
+    local position
+    if(centered) then 
+        position = getWidgetCentre(width, self.y - y)
+    else
+        position = GuiDim.new(0, (1920 / 2 - width / 2) + x, 0, self.y - y)
+    end
+    self.y = (self.y + y) - self.widgetGap
     return position
 end
 
+function StackMenu:initBasicWidget(widget, x, y, w, h, keepAlignment, texture, textSize)
```

Comment:
Not sure how I feel about this change. 

The idea of the stack menu is the widgets are always centered, if they need not be then the widget is able to be set without the menu, eg line 34 to 43 here https://github.com/Hopson97/open-builder/blob/master/game/client/menus/world_select.lua#L34 the "join world" button needed to be at the top-right of the window, so I didn't use the stack menu.

For checkbox I guess is a special case though.

I slightly different solution would be to call the function as normal like the function was before, and then manually set the position afterward to reset it, eg:

```lua
function StackMenu:addCheckBox(text)
    local checkBox = self.overlay:addCheckBox()
    self:initBasicWidget(checkBox)
    checkbox.position = GuiDim.new ( etc etc)
    checkBox.text = text
    return checkBox
end
````

This way the rest of the functions in the file are able to stay the same as they were before, rather than adapting to a new widget being added to the menu helper.

Reply:
Hmm, I do completely see your point. I'll get to work implementing that on my test version so I can have that ready to go soon. Just checking then, should I scrap the whole more generalized initialization function then? As described in my general comment all the way down at the bottom of this page? 

Hope I'm not coming off as pushy in any way, I'm just here to help :) 

Thanks for all your time!